avocado/utils/softwareraid.py

```diff
@@ -0,0 +1,161 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+# Author: Venkat Rao B <vrbagal1@linux.vnet.ibm.com>
+
+
+import time
+import logging
+from . import process
+
+log = logging.getLogger('avocado.test')
+
+
+class SoftwareRaid:
+    """
+    Perform software raid related operations.
+
+    :param name: Name of the software raid to be created
+    :type name: str
+    :param level: Level of software raid to be created
+    :type name: str
+    :param disk: List of disks for software raid
+    :type disk: list
+    :param metadata: Metadata level for software raid
+    :type metadata: str
+    :param spare: List of spare disks for software raid
+    :type spare: list
+    """
+    def __init__(self, name, level, disk, metadata, spare=''):
+        self.name = name
+        self.level = level
+        self.metadata = metadata
+        self.disk = disk
+        self.remadd = ''
+        self.spare = spare
+
+    def create(self):
+        """
+        Creates software raid.
+
+        :return: True if raid is created. False otherwise.
+        :rtype: bool
+        """
+        cmd = "mdadm --create --assume-clean %s" % self.name
+        cmd += " --level=%s" % self.level
+        cmd += " --raid-devices=%d %s" % (len(self.disk),
+                                          " ".join(self.disk))
+        cmd += " --metadata=%s" % self.metadata
+        if self.spare:
+            cmd += " --spare-devices=%d %s" % (len(self.spare),
+                                               " ".join(self.spare))
+        cmd += " --verbose --force"
+        return self._run_command(cmd)
+
+    def _run_command(self, cmd, log_details=True):
+        if process.system(cmd, ignore_status=True, shell=True) != 0:
+            if log_details:
+                log.debug(self.get_detail())
+            return False
+        if log_details:
+            log.debug(self.get_detail())
+        return True
+
+    def get_detail(self):
+        """
+        Returns mdadm details.
+
+        :return: mdadm --detail output
+        :rtype: str
+        """
+        cmd = "mdadm --detail %s" % self.name
+        output = process.run(cmd, ignore_status=True, shell=True)
```

Comment:
You have a method named `_run_command` and sometimes you call `process.run()` directly. It would be nice to call all the commands on the same way.

Reply:
@beraldoleal I can remove from every other places, but can not remove from `get_detail()` here because `_run_command()` calls this method when logging.